File: clients/ruby/exampleCheck.rb
#!/usr/bin/env ruby

require 'json'
require 'net/http'
require 'uri'

# reporting_url fetches the Kuberhealthy reporting endpoint from the environment.
def reporting_url
  url = ENV['KH_REPORTING_URL']
  return url if url && !url.empty?
  raise 'KH_REPORTING_URL not set'
end

# run_uuid fetches the unique run identifier from the environment.
def run_uuid
  uuid = ENV['KH_RUN_UUID']
  return uuid if uuid && !uuid.empty?
  raise 'KH_RUN_UUID not set'
end

# post_status sends a status report back to Kuberhealthy.
def post_status(ok:, errors: [])
  uri = URI(reporting_url)
  req = Net::HTTP::Post.new(uri)
  req['Content-Type'] = 'application/json'
  req['kh-run-uuid'] = run_uuid
  req.body = JSON.generate({ ok: ok, errors: errors })
  Net::HTTP.start(uri.hostname, uri.port) { |http| http.request(req) }
end

# report_success tells Kuberhealthy the check passed.
def report_success
  post_status(ok: true, errors: [])
end

# report_failure tells Kuberhealthy the check failed with messages.
def report_failure(messages)
  post_status(ok: false, errors: messages)
end

begin
  # TODO: add check logic here. For now, always succeed.
  report_success
rescue StandardError => e
  report_failure([e.message])
end

File: clients/java/ExampleCheck.java
import java.io.IOException;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class ExampleCheck {
    public static void main(String[] args) throws Exception {
        String url = System.getenv("KH_REPORTING_URL");
        String uuid = System.getenv("KH_RUN_UUID");

        if (url == null || url.isEmpty() || uuid == null || uuid.isEmpty()) {
            System.err.println("KH_REPORTING_URL and KH_RUN_UUID must be set");
            System.exit(1);
        }

        // Replace this with your own check logic. If the check passes, leave
        // ok as true. To report a failure, set ok to false and provide an
        // error message.
        boolean ok = true;
        String error = "";

        // Example: report a failure with an error message
        // ok = false;
        // error = "something went wrong";

        report(url, uuid, ok, error);
    }

    private static void report(String url, String uuid, boolean ok, String errorMessage) throws IOException {
        String payload = "{\"ok\":true,\"errors\":[]}";
        if (!ok) {
            payload = "{\"ok\":false,\"errors\":[\"" + escape(errorMessage) + "\"]}";
        }

        HttpURLConnection conn = (HttpURLConnection) new URL(url).openConnection();
        conn.setRequestMethod("POST");
        conn.setRequestProperty("Content-Type", "application/json");
        conn.setRequestProperty("kh-run-uuid", uuid);
        conn.setDoOutput(true);

        try (OutputStream os = conn.getOutputStream()) {
            os.write(payload.getBytes(StandardCharsets.UTF_8));
        }

        int code = conn.getResponseCode();
        if (code != 200) {
            // Kuberhealthy rejects malformed reports with HTTP 400. Treat any
            // non-200 status as a failure so the container exits quickly.
            throw new IOException("unexpected response code: " + code);
        }
    }

    private static String escape(String s) {
        return s.replace("\\", "\\\\").replace("\"", "\\\"");
    }
}
